stixcore/util/scripts/end2end_testing.py

```diff
@@ -0,0 +1,150 @@
+    "L1/2020/06/16/QL/solo_L1_stix-ql-variance_20200616_V01.fits",
+    "L1/2021/11/16/QL/solo_L1_stix-ql-spectra_20211116_V01.fits",
+    "L1/2021/11/16/CAL/solo_L1_stix-cal-energy_20211116_V01.fits",
+    # HK
+    "L1/2020/06/16/HK/solo_L1_stix-hk-maxi_20200616_V01.fits",
+    "L1/2021/09/20/HK/solo_L1_stix-hk-mini_20210920_V01.fits"]
+
+remote = ["http://pub099.cs.technik.fhnw.ch/data/fits_test/" + x for x in END_TO_END_TEST_FILES]
+# files = ["/home/shane/fits_test/" + x for x in files]
+files = [("/home/shane/fits_test_latest/" + x, remote[i])
+         for i, x in enumerate(END_TO_END_TEST_FILES)]
+
+
+def _pretty_xml_print(current, parent=None, index=-1, depth=0):
+    for i, node in enumerate(current):
+        _pretty_xml_print(node, current, i, depth + 1)
+    if parent is not None:
+        if index == 0:
+            parent.text = '\n' + ('\t' * depth)
+        else:
+            parent[index - 1].tail = '\n' + ('\t' * depth)
+        if index == len(parent) - 1:
+            current.tail = '\n' + ('\t' * (depth - 1))
+
+
+def rebuild_end2end(files, *, splits=3, socdir=Path("/home/shane/tm/"),
+                    outdir=test_data.products.DIREND2END):
+    splitfiles = list(range(0, splits))
+    packetid = 0
+    openxml = ""
+    rootxml = None
+
+    outdir.mkdir(parents=True, exist_ok=True)
+
+    newroot = [Et.Element('Response') for i in splitfiles]
+    newresponse = [Et.SubElement(newroot[i], 'PktRawResponse') for i in splitfiles]
+
+    for f, u in files:
+        try:
+            l1_f = Path(f)
+            l1_p = Product(l1_f)
+            rootdir = l1_f.parent.parent.parent.parent.parent.parent
+
+            l0_f = (rootdir / "L0").rglob(l1_p.parent[0]).__next__()
+            l0_p = Product(l0_f)
+
+            lb_f = (rootdir / "LB").rglob(l0_p.parent[0]).__next__()
+            lb_p = Product(lb_f)
+
+            for row in lb_p.control:
+                # print(f"{row['raw_file']} > {row['packet']}")
+                if openxml != row['raw_file']:
+                    rootxml = Et.parse(str(socdir / row['raw_file'])).getroot()
+                    openxml = row['raw_file']
+                    PktRawResponseElement = dict()
+                    for node in rootxml.iter('PktRawResponseElement'):
+                        PktRawResponseElement[str(node.attrib['packetID'])] = node
+
+                node = PktRawResponseElement[str(row['packet'])]
+                # print(f"found: {node}")
+                packetid += 1
+                rpa = Et.SubElement(newresponse[packetid % splits], "PktRawResponseElement")
+                rpa.set("packetID", str(packetid))
+                pa = Et.SubElement(rpa, "packet")
+                pa.text = list(node)[0].text
+        except Exception as e:
+            print(e)
+    print(f"{packetid} packets recompiled")
+
+    for i, root in enumerate(newroot):
+        with open(outdir / f'Test{i}.PktTmRaw.xml', 'wb') as xmlf:
+            _pretty_xml_print(root)
```

Comment:
lxml has a pretty print function but maybe it not what you need `etree.tostring(x, pretty_print=True)`